crates/common/mqtt_channel/src/tests.rs

```diff
@@ -429,4 +436,20 @@ mod tests {
         assert!(result.is_err());
         assert!(result.unwrap_err().to_string().contains("Invalid session"));
     }
+
+    #[tokio::test]
+    #[serial]
+    async fn subscription_failures() {
+        let broker = mqtt_tests::test_mqtt_broker();
+        let mqtt_config = Config::default().with_port(broker.port);
+
+        let topic = TopicFilter::new_unchecked("test/topic");
+        let mqtt_config = mqtt_config.with_subscriptions(topic);
+
+        if let Err(MqttError::SubscriptionFailure) = Connection::new(&mqtt_config).await {
```

Comment:
As a suggestion, maybe one could write it in this form:

```rust
assert!(matches!(Connection::new(&mqtt_config).await, Err(MqttError::SubscriptionFailure)), "Expected the connection to fail.")
```

Reply:
Haha, I was thinking the same thing!

Reply:
Updated.